### PR TITLE
Add Unit Test Cases for Neural Network Initialization 

### PR DESCRIPTION
**Pull request by jijoongmoon**
Nerual Network Initialization Unit Test including layer
initialization.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/src/neuralnet.cpp [hunk 1]
```diff
@@ -255,6 +263,7 @@ int NeuralNetwork::init() {
 
   if (ini == NULL) {
     ml_loge("Error: cannot parse file: %s\n", ini_file.c_str());
+    iniparser_freedict(ini);
```
Comment (dongju-chae) [inline nntrainer/src/neuralnet.cpp:266]: freedict() for the null ini instance?

File: nntrainer/src/layers.cpp [hunk 2]
```diff
@@ -141,17 +151,30 @@ Tensor InputLayer::forwarding(Tensor in) {
   return input;
 }
 
-void InputLayer::initialize(int b, int h, int w, int id, bool init_zero,
-                            WeightIniType wini) {
+int InputLayer::initialize(int b, int h, int w, int id, bool init_zero,
+                           WeightIniType wini) {
+  int status = ML_ERROR_NONE;
+  if (b <= 0 || h <= 0 || w <= 0) {
```
Comment (dongju-chae) [inline nntrainer/src/layers.cpp:157]: I think it's better for those variables to be unsigned if they are always positive (or zero) values.

File: nntrainer/src/databuffer.cpp [hunk 3]
```diff
@@ -425,24 +425,37 @@ bool DataBuffer::getDataFromBuffer(
...
+    if (!data_file.good()) {
+      ml_loge(
+        "Error: Cannot open data file, Datafile is necessary for training");
+      return ML_ERROR_INVALID_PARAMETER;
+    }
     train_file = path;
     break;
   case DATA_VAL:
+    if (!data_file.good()) {
+      ml_logw("Waring: Cannot open validation data file. Cannot validate "
```
Comment (zhoonit) [inline nntrainer/src/databuffer.cpp:441]: I think it is typo `Waring` -> `Warning` and also for line 449

File: nntrainer/src/neuralnet.cpp (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/src/neuralnet.cpp:266]: fixed. Thanks.!

File: nntrainer/src/layers.cpp (hunk 2 above)
Comment (jijoongmoon) [reply nntrainer/src/layers.cpp:157]: Yes. I will change them later commit. Thanks.

File: nntrainer/src/databuffer.cpp (hunk 3 above)
Comment (jijoongmoon) [reply nntrainer/src/databuffer.cpp:441]: Thanks.!

**Pull request merged**